From 8c3841eb07fac93fa12cc56ffa2640e4ca7f10a4 Mon Sep 17 00:00:00 2001
From: Lionel VITTE <lionel.vitte@st.com>
Date: Thu, 21 Mar 2024 12:55:02 +0100
Subject: [PATCH 0902/1695] checkpatch: discard line starting with Change-Id

In commit message, there could be a Change-Id information
if Gerrit is used. In some case, the Change-Id number can
be similar to a word and produce codespell warnings.
For example
  WARNING: 'daa' may be misspelled - perhaps 'data'?
  #9:
  Change-Id: I20dc18f57e4daa3168d481d4de08c138dc2eaa80
So discard such line

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 90e235d4c24e9969013f870b7e339ec9673addff checkpatch: discard line starting with Change-Id]
Change-Id: I046d30b3953616cd0b456afccbbbad0b328ee357
Signed-off-by: Lionel VITTE <lionel.vitte@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/368305
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
(cherry picked from commit 44ad3cbb9b0a5a369ad2a11c48960da0be1c142c)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/387160
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 scripts/checkpatch_inc.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/checkpatch_inc.sh b/scripts/checkpatch_inc.sh
index 83c6829a7..a6fbb630a 100644
--- a/scripts/checkpatch_inc.sh
+++ b/scripts/checkpatch_inc.sh
@@ -32,7 +32,8 @@ function checkpatch() {
 	# The first git 'format-patch' shows the commit message
 	# The second one produces the diff (might be empty if _CP_EXCL
 	# filters out all diffs)
-	(git format-patch $1^..$1 --stdout | sed -n '/^diff --git/q;p'; \
+	(git format-patch $1^..$1 --stdout | sed -n '/^diff --git/q;p' | \
+	 sed '/^Change-Id: /d'; \
 	 git format-patch $1^..$1 --stdout -- $_CP_EXCL . | \
 		sed -n '/^diff --git/,$p') | _checkpatch
 }
-- 
2.47.2

